kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/resources/kms/aws/KubeAwsKmsTestKmsFacade.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.systemtests.resources.kms.aws;
+
+import java.io.IOException;
+import java.net.URI;
+import java.time.Duration;
+import java.util.List;
+import java.util.Objects;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.kroxylicious.kms.provider.aws.kms.AbstractAwsKmsTestKmsFacade;
+import io.kroxylicious.kms.provider.aws.kms.model.DescribeKeyResponse;
+import io.kroxylicious.kms.service.TestKekManager;
+import io.kroxylicious.kms.service.UnknownAliasException;
+import io.kroxylicious.systemtests.Environment;
+import io.kroxylicious.systemtests.executor.Exec;
+import io.kroxylicious.systemtests.executor.ExecResult;
+import io.kroxylicious.systemtests.installation.kms.aws.AwsKms;
+import io.kroxylicious.systemtests.installation.kms.aws.AwsKmsClient;
+import io.kroxylicious.systemtests.installation.kms.aws.AwsKmsLocal;
+import io.kroxylicious.systemtests.k8s.exception.KubeClusterException;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+import static io.kroxylicious.kms.provider.aws.kms.AwsKms.ALIAS_PREFIX;
+import static io.kroxylicious.kms.provider.aws.kms.AwsKmsTestKmsFacade.CREATE_KEY_RESPONSE_TYPE_REF;
+import static io.kroxylicious.kms.provider.aws.kms.AwsKmsTestKmsFacade.DESCRIBE_KEY_RESPONSE_TYPE_REF;
+import static io.kroxylicious.kms.provider.aws.kms.AwsKmsTestKmsFacade.SCHEDULE_KEY_DELETION_RESPONSE_TYPE_REF;
+import static io.kroxylicious.systemtests.k8s.KubeClusterResource.cmdKubeClient;
+
+/**
+ * KMS Facade for AWS Kms running inside Kube.
+ * Uses command line interaction so to avoid the complication of exposing the AWS endpoint
+ * to the test outside the cluster.
+ */
+public class KubeAwsKmsTestKmsFacade extends AbstractAwsKmsTestKmsFacade {
```

Comment:
You've given the class the responsibility to function with both AWS Cloud and AWS Localstack, but the name doesn't convey this.